### PR TITLE
Feature/delete

### PR DESCRIPTION
**Pull request by MisterAJ-Prodigy**
Adds the ability to delete todos.

### CODE REVIEW

File: assets/scripts/views/TodoAppView.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
         'submit .js-todo-form': 'addTodo',
+        'change .js-todo-delete-checkbox': 'deleteTodo',
     },
 
     initialize(options) {
         this.template = Handlebars.compile($('#todo-app-template').html());
-        this.collection.add({ title: 'Hello Trevor' });
+        this.collection.add({
+            title: 'Hello Trevor',
+            created_at: Date.now(),
```
Comment (trevorwright) [inline assets/scripts/views/TodoAppView.js]: Consistency, `createAt`, no snake case.

File: assets/scripts/views/TodoAppView.js [hunk 2]
```diff
@@ -20,6 +25,7 @@ app.TodoAppView = Backbone.View.extend({
 
     setListener() {
         this.listenTo(this.collection, 'add', this.renderChild);
+        this.listenTo(this.collection, 'remove', this.renderChild);
```
Comment (trevorwright) [inline assets/scripts/views/TodoAppView.js]: Merge this with the line above it, don't need two separate ones

File: assets/scripts/views/TodoAppView.js [hunk 3]
```diff
@@ -36,8 +42,16 @@ app.TodoAppView = Backbone.View.extend({
         const $input = this.$('.js-todo-input')[0];
         event.preventDefault();
         if ($input.value.length > 0) {
-            this.collection.add({ title: $input.value });
+            this.collection.add({
+                title: $input.value,
+                created_at: Date.now(),
```
Comment (trevorwright) [inline assets/scripts/views/TodoAppView.js]: both created at, and id should be automated in the initialize, someone shouldn't need to create these on their own.

This is also a weird hack for determining Ids.

File: assets/scripts/views/TodoView.js [hunk 4]
```diff
@@ -16,6 +16,14 @@ app.TodoView = Backbone.View.extend({
     toggleDoneClass(event) {
         const isChecked = this.$('.js-todo-done-checkbox').prop('checked');
         this.$('.js-todo-title').toggleClass('todo-list__item-title--done', isChecked);
+        this.showDeleteBox(isChecked);
         this.model.set('completed', isChecked);
     },
+
+    showDeleteBox(isChecked) {
+        this.$('.js-todo-delete-checkbox')
```
Comment (trevorwright) [inline assets/scripts/views/TodoView.js]: Delete as a checkbox is really odd, probably a button or something makes more sense.

File: assets/scripts/views/TodoView.js [hunk 5]
```diff
@@ -16,6 +16,14 @@ app.TodoView = Backbone.View.extend({
...
+            .toggleClass('todo-list__item-checkbox--hidden', !isChecked);
```
Comment (trevorwright) [inline assets/scripts/views/TodoView.js]: Why would this get hidden, if its deleted doesn't the item get removed?

File: index.html [hunk 6]
```diff
@@ -41,6 +41,8 @@ <h1 class='todo-main__header'>Aj's Magical Todo Maker!!</h1>
             <h3 class="todo-list__item-title js-todo-title">{{ title }}</h3>
             <label for="todo-list__item-checkbox ">Done?</label>
             <input type="checkbox" class="todo-list__item-checkbox js-todo-done-checkbox"/>
+            <label for="todo-list__item-checkbox" class="todo-list__item-checkbox--hidden js-todo-delete-label">Delete?</label>
```
Comment (trevorwright) [inline index.html]: So if these are hidden, what prompts them to show up?

File: assets/scripts/views/TodoView.js (hunk 5 above)
Comment (MisterAJ-Prodigy) [reply assets/scripts/views/TodoView.js]: It's hidden until you click done

File: index.html (hunk 6 above)
Comment (MisterAJ-Prodigy) [reply index.html]: clicking Done

Comment (trevorwright) [reply index.html]: Done is a delete? This sounds confusing.

Keep in mind, this is the implementation you were aiming for. Style wise I'm not too worried about copying it exactly, but follow the functionality.

http://todomvc.com/examples/react/#/

File: assets/scripts/views/TodoView.js [hunk 7]
```diff
@@ -5,6 +5,8 @@ app.TodoView = Backbone.View.extend({
 
     initialize() {
         this.template = Handlebars.compile($('#todo-item-template').html());
+        this.model.set('id', this.model.cid);
```
Comment (trevorwright) [inline assets/scripts/views/TodoView.js]: So one thing you'll want to try to do is bunch up these `model.set` calls. If the view is listening to `change` the event will actually get fired twice for each of these lines.

Luckily backbone has support for this!

```
this.model.set({
  id: this.model.cid,
  createdAt: Date.now(),
});
```

**Pull request merged**